local-interchaintest/src/utils/manager.rs

```diff
@@ -0,0 +1,210 @@
+    code_ids_map.insert(REVERSE_SPLITTER_NAME.to_string(), reverse_splitter_code_id);
+    code_ids_map.insert(FORWARDER_NAME.to_string(), forwarder_code_id);
+
+    gc.contracts
+        .code_ids
+        .insert(NEUTRON_CHAIN_NAME.to_string(), code_ids_map);
+
+    // init the registry
+    let registry_init_msg = valence_workflow_registry_utils::InstantiateMsg {
+        admin: NEUTRON_CHAIN_ADMIN_ADDR.to_string(),
+    };
+
+    let registry = contract_instantiate(
+        test_ctx
+            .get_request_builder()
+            .get_request_builder(NEUTRON_CHAIN_NAME),
+        DEFAULT_KEY,
+        registry_code_id,
+        &serde_json::to_string(&registry_init_msg).unwrap(),
+        "workflow-registry",
+        None,
+        "",
+    )
+    .unwrap();
+
+    gc.general.registry_addr = registry.address;
+
+    // let code_ids = GLOBAL_CONFIG.read().unwrap().contracts.code_ids.clone();
+    // println!("{:?}", code_ids);
+
+    // read the log file
+    // setup the chain info (and later bridge info) for the global config
+    // upload contracts to the chains and get their ids, and update the global config
+    // upload and instantiate a registry (for the manager) and update the global config
+
+    Ok(())
+}
+
+/// Get the chain infos and bridge info from the log file
+pub fn get_data_from_log() -> HashMap<String, ChainInfo> {
+    let log_file = fs::File::open("local-interchaintest/configs/logs.json")
+        .expect("file should open read only");
+    let log_json: serde_json::Value =
+        serde_json::from_reader(log_file).expect("file should be proper JSON");
+
+    let mut chain_infos = HashMap::new();
+
+    log_json
+        .get("chains")
+        .expect("file should have chains")
+        .as_array()
+        .expect("Chains should be an array")
+        .iter()
+        // TODO: change to map later for all chains we have
+        .for_each(|chain_data| {
+            let chain_name = chain_data
+                .get("chain_name")
+                .expect("Chain data must have chain_name")
+                .as_str()
+                .expect("chain_name must be a string")
+                .replace("local", "");
+
+            // TODO: We only want neutron for now, change later hardcoded values
+            if !chain_name.contains("neutron") {
+                return;
+            }
+
+            let rpc = chain_data
+                .get("rpc_address")
+                .expect("Chain data must have rpc_address")
+                .as_str()
+                .expect("rpc_address must be a string");
+            let grpc = chain_data
+                .get("grpc_address")
+                .expect("Chain data must have grpc_address")
+                .as_str()
+                .expect("grpc_address must be a string");
+
+            chain_infos.insert(
+                NEUTRON_CHAIN_NAME.to_string(),
+                ChainInfo {
+                    name: NEUTRON_CHAIN_NAME.to_string(),
```

Comment:
we could use chain_name here then we can add all of them already

Reply:
Neutron is a special case, and currently only wanted to set up everything for a single chain
Will look into it in the multi-chain manager